README.md

```diff
@@ -1 +1,121 @@
+
+* `body: bytes`
+* `content_types: Optional[Tuple[bytes]], default = None`
+* `http_origin: bool, default = True`
+* `no_sniff: bool, default = False`
+* `extra_types: Optional[Tuple[Tuple[bytes, bytes, Optional[Set[bytes]], bytes], ...]], default = None`
+* `supported_types: Set[bytes], default = None`
+
+Return the [MIME type essence](https://mimesniff.spec.whatwg.org/#mime-type-essence) (e.g. `text/html`) matching the input data, or 
+`None` if no match can be found.
+
+The `body` parameter is the byte sequence of which MIME type is to be determined. `xtractmime` only considers the first few
+bytes of the `body` and the specific number of bytes read is defined in the `xtractmime.RESOURCE_HEADER_BUFFER_LENGTH` constant.
+
+`content_types` is a tuple of MIME types given in the resource metadata. For resources retrieved via HTTP, users should pass the list of MIME types mentioned in `Content-Type` header. Otherwise pass the MIME type indicated for the resource.
+
+`http_origin` indicates if the resource has been retrieved via HTTP (`True`, default) or not (`False`).
+
+`no_sniff` is a flag which is *`True`* if the user agent does not wish to
+perform sniffing on the resource and *`False`* (by default) otherwise. Users may want to set
+this parameter to *`True`* if the [`X-Content-Type-Options`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options) response header is set to `nosniff`. For more info, see [here](https://mimesniff.spec.whatwg.org/#no-sniff-flag).
+
+Optional `supported_types` is a set of all [MIME types supported the by user agent](https://mimesniff.spec.whatwg.org/#supported-by-the-user-agent) and if `supported_types` is not
+specified, all MIME types are assumed to be supported. Using this parameter can improve the performance of `xtractmime`.
+
+`extra_types` is a tuple of patterns to support detecting additional MIME types. Each entry in the tuple should follow the format
+**(Byte Pattern, Pattern Mask, Leading Bytes, MIME type)**. 
+
+* **Byte Pattern** is a byte sequence to compare with the first few bytes (``xtractmime.RESOURCE_HEADER_BUFFER_LENGTH``) of the `body`.
+* **Pattern Mask** is a byte sequence that indicates the significance of **Byte Pattern** bytes: `b"\xff"` indicates the matching byte is strictly significant, `b"\xdf"` indicates that the byte is significant in an ASCII case-insensitive way, and `b"\x00"` indicates that the byte is not significant.
+* **Leading Bytes** is a set of bytes to be ignored while matching the leading bytes in the content.
+* **MIME type** should be returned if the pattern matches.
+
+**Sample `extra_types`:**
+```python
+extra_types = ((b'test', b'\xff\xff\xff\xff', None, b'text/test'),...)
+```
+
+---
+**NOTE**
+
+*Users should be careful while using `extra_types` argument as it may introduce some privilege escalation vulnerabilities for `xtractmime`. For more info, see [here](https://mimesniff.spec.whatwg.org/#ref-for-mime-type%E2%91%A1%E2%91%A8).*
```

Comment:
```suggestion
*Be careful while using the `extra_types` argument, as it may introduce some privilege escalation vulnerabilities for `xtractmime`. For more info, see [here](https://mimesniff.spec.whatwg.org/#ref-for-mime-type%E2%91%A1%E2%91%A8).*
```